airsonic-main/src/main/java/org/airsonic/player/dao/AbstractDao.java

```diff
@@ -69,11 +73,19 @@ protected String prefix(String columns, String prefix) {
         l.replaceAll(s -> prefix + "." + s);
         return String.join(", ", l);
     }
+    
+    protected static Object[] convertArgs(Object[] args) {
+        return args == null ? null : Stream.of(args).map(x -> (Object) ((x instanceof Instant) ? Timestamp.from((Instant) x) : x)).collect(Collectors.toList()).toArray();
+    }
+    
+    protected static Map<String, Object> convertArgs(Map<String, Object> args) {
+        return args == null ? null : args.entrySet().parallelStream().map(x -> (x.getValue() instanceof Instant) ? Pair.of(x.getKey(), Timestamp.from((Instant) x.getValue())) : x).collect(Collectors.toMap(x -> x.getKey(), x -> x.getValue()));
```

Comment:
I do think it would be cleaner to have a separate method for converting single db objects... Then you would have to use a big lambda inside the `map`, it's hard to read.

Reply:
The way I view it, I think most of this is temporary plumbing code until we get (a) a more recent jdbc compliant driver (than hsqldb 1.8) that supports these more recent additions out of the box, and (b) move to a more pliable sql management library that'll do most of this work for us instead of relying on jdbc templates.

This stuff is here so people who want to remain on older versions can also enjoy the benefits, thus we need to provide these adapters to bridge them between old and new. As such, short and succinct works well enough, without putting more effort in.

I'll separate it out into multiple lines for readability, but to my lambdafied eyes, it's a simple stream function that checks every map entry to see if the value is an Instant, and if it is, it converts it to a Timestamp, and reassociates it with the same key. Otherwise it keeps the entry as is.

Reply:
Ah, I did not know about this being needed because of HSQLDB 1.8. Then let's add a `FIXME` comment so that we remember why we need this right now!